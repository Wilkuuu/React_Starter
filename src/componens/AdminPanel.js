import React from "react";

class AdminPanel extends React.Component {

  constructor() {
    super();
    this.state ={
      book :{
        name : "",
        author: "",
        description: "",
        onStock: false,
        image: ""
        
        
      }
    };

  };

  handleChange = (event) => {

    let newBook;

    if(event.target.name === "onStock") {
       newBook = {
         ...this.state.book,
         [event.target.name]: event.target.checked
        
       };
    } else {

    newBook = {
          ...this.state.book,
         [event.target.name]: event.target.value
    };
  }
   this.setState({
     book: newBook
   });
  

  }



  render() {
    const checkboxColor = {
      color: "white"
    };

    return (
      <div className="adminpanel col-md-4">
        <form>
          <div className="form-group">
            <input
              type="text"
              placeholder="Nazwa ksiazki"
              id="name"
              name="name"
              className="form-control"
              onChange={this.handleChange}  value={this.state.book.name}
            />
          </div>
          <div className="form-group">
            <input
              type="text"
              placeholder="Autor"
              id="author"
              name="author"
              className="form-control"
              onChange={this.handleChange}  value={this.state.book.author}
            />
          </div>
          <div className="form-group">
            <textarea
              placeholder="Opis książki"
              id="description"
              name="description"
              className="form-control"
              onChange={this.handleChange}  value={this.state.book.description}
            />
          </div>
          <div className="form-group">
            <label
              htmlFor="onStock"
              className="form-check-label"
              style={checkboxColor}
             
            >
              Dostępność:                   {" "}
            </label>
            <hr></hr>
            <input
              type="checkbox"
              id="onStock"
              name="onStock"
              className="form-check-input"
              onChange={this.handleChange}  value={this.state.book.onStock}
            />
          </div>
          <div className="form-group">
            <textarea
              placeholder="Grafika"
              id="image"
              name="image"
              className="form-control"
              onChange={this.handleChange}  value={this.state.book.image}
            />
          </div>
          <button type="submit" className="btn btn-success">
            Dodaj
          </button>
          </form>
      </div>
    );
  }
}

export default AdminPanel;
