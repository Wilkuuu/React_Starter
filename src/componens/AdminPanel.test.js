import React from 'react';
import ReactDOM from 'react-dom';
import AdminPanel from './AdminPanel';

import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

describe('Admin Panel tests', () => {

    it('Adminpanel render correct', () => { 
    const div = document.createElement('div');
    ReactDOM.render(<AdminPanel/>, div);
    ReactDOM.unmountComponentAtNode(div);
    })

   it('Panel wyrenderowano', () => {
       const wrapper = shallow(<AdminPanel/>);
       expect(wrapper).toMatchSnapshot();
   })


})