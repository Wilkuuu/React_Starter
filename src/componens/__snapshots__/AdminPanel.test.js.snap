// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admin Panel tests Panel wyrenderowano 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AdminPanel />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="name"
            name="name"
            placeholder="Nazwa ksiazki"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <input
            className="form-control"
            id="author"
            name="author"
            placeholder="Autor"
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            id="description"
            name="description"
            placeholder="Opis książki"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            className="form-check-label"
            htmlFor="bookOnStock"
            style={
              Object {
                "color": "white",
              }
            }
          >
            Dostępność:
             
          </label>
          <input
            className="form-check"
            id="bookOnStock"
            name="bookOnStock"
            placeholder="Dostęność książki"
            type="checkbox"
          />
        </div>
        <div
          className="form-group"
        >
          <textarea
            className="form-control"
            id="image"
            name="image"
            placeholder="Grafika"
          />
        </div>
        <button
          className="btn btn-success"
          type="submit"
        >
          Dodaj
        </button>
      </form>,
      "className": "adminpanel col-md-4",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Nazwa ksiazki"
              type="text"
            />
          </div>,
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Autor"
              type="text"
            />
          </div>,
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Opis książki"
            />
          </div>,
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
              htmlFor="bookOnStock"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Dostępność:
               
            </label>
            <input
              className="form-check"
              id="bookOnStock"
              name="bookOnStock"
              placeholder="Dostęność książki"
              type="checkbox"
            />
          </div>,
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="image"
              name="image"
              placeholder="Grafika"
            />
          </div>,
          <button
            className="btn btn-success"
            type="submit"
          >
            Dodaj
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Nazwa ksiazki"
              type="text"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "name",
              "name": "name",
              "placeholder": "Nazwa ksiazki",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Autor"
              type="text"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "author",
              "name": "author",
              "placeholder": "Autor",
              "type": "text",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Opis książki"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "description",
              "name": "description",
              "placeholder": "Opis książki",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="form-check-label"
                htmlFor="bookOnStock"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Dostępność:
                 
              </label>,
              <input
                className="form-check"
                id="bookOnStock"
                name="bookOnStock"
                placeholder="Dostęność książki"
                type="checkbox"
              />,
            ],
            "className": "form-group",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Dostępność:",
                  " ",
                ],
                "className": "form-check-label",
                "htmlFor": "bookOnStock",
                "style": Object {
                  "color": "white",
                },
              },
              "ref": null,
              "rendered": Array [
                "Dostępność:",
                " ",
              ],
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-check",
                "id": "bookOnStock",
                "name": "bookOnStock",
                "placeholder": "Dostęność książki",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <textarea
              className="form-control"
              id="image"
              name="image"
              placeholder="Grafika"
            />,
            "className": "form-group",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "form-control",
              "id": "image",
              "name": "image",
              "placeholder": "Grafika",
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Dodaj",
            "className": "btn btn-success",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Dodaj",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="name"
              name="name"
              placeholder="Nazwa ksiazki"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <input
              className="form-control"
              id="author"
              name="author"
              placeholder="Autor"
              type="text"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="description"
              name="description"
              placeholder="Opis książki"
            />
          </div>
          <div
            className="form-group"
          >
            <label
              className="form-check-label"
              htmlFor="bookOnStock"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Dostępność:
               
            </label>
            <input
              className="form-check"
              id="bookOnStock"
              name="bookOnStock"
              placeholder="Dostęność książki"
              type="checkbox"
            />
          </div>
          <div
            className="form-group"
          >
            <textarea
              className="form-control"
              id="image"
              name="image"
              placeholder="Grafika"
            />
          </div>
          <button
            className="btn btn-success"
            type="submit"
          >
            Dodaj
          </button>
        </form>,
        "className": "adminpanel col-md-4",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="name"
                name="name"
                placeholder="Nazwa ksiazki"
                type="text"
              />
            </div>,
            <div
              className="form-group"
            >
              <input
                className="form-control"
                id="author"
                name="author"
                placeholder="Autor"
                type="text"
              />
            </div>,
            <div
              className="form-group"
            >
              <textarea
                className="form-control"
                id="description"
                name="description"
                placeholder="Opis książki"
              />
            </div>,
            <div
              className="form-group"
            >
              <label
                className="form-check-label"
                htmlFor="bookOnStock"
                style={
                  Object {
                    "color": "white",
                  }
                }
              >
                Dostępność:
                 
              </label>
              <input
                className="form-check"
                id="bookOnStock"
                name="bookOnStock"
                placeholder="Dostęność książki"
                type="checkbox"
              />
            </div>,
            <div
              className="form-group"
            >
              <textarea
                className="form-control"
                id="image"
                name="image"
                placeholder="Grafika"
              />
            </div>,
            <button
              className="btn btn-success"
              type="submit"
            >
              Dodaj
            </button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="name"
                name="name"
                placeholder="Nazwa ksiazki"
                type="text"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "name",
                "name": "name",
                "placeholder": "Nazwa ksiazki",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <input
                className="form-control"
                id="author"
                name="author"
                placeholder="Autor"
                type="text"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "author",
                "name": "author",
                "placeholder": "Autor",
                "type": "text",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="form-control"
                id="description"
                name="description"
                placeholder="Opis książki"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "description",
                "name": "description",
                "placeholder": "Opis książki",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <label
                  className="form-check-label"
                  htmlFor="bookOnStock"
                  style={
                    Object {
                      "color": "white",
                    }
                  }
                >
                  Dostępność:
                   
                </label>,
                <input
                  className="form-check"
                  id="bookOnStock"
                  name="bookOnStock"
                  placeholder="Dostęność książki"
                  type="checkbox"
                />,
              ],
              "className": "form-group",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Dostępność:",
                    " ",
                  ],
                  "className": "form-check-label",
                  "htmlFor": "bookOnStock",
                  "style": Object {
                    "color": "white",
                  },
                },
                "ref": null,
                "rendered": Array [
                  "Dostępność:",
                  " ",
                ],
                "type": "label",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "form-check",
                  "id": "bookOnStock",
                  "name": "bookOnStock",
                  "placeholder": "Dostęność książki",
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <textarea
                className="form-control"
                id="image"
                name="image"
                placeholder="Grafika"
              />,
              "className": "form-group",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "form-control",
                "id": "image",
                "name": "image",
                "placeholder": "Grafika",
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Dodaj",
              "className": "btn btn-success",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Dodaj",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
